files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java

```diff
@@ -739,6 +745,290 @@ public Mono<FileDetail> create(String clientCode, String uri, FilePart fp, Strin
+
+			BufferedImage flippedImage = resizedImage;
+			if (imageDetails.getFlipHorizontal().booleanValue()) {
+				BufferedImage flippedHorizontalImage = flipHorizontal(flippedImage);
+				flippedImage = null;
+				flippedImage = flippedHorizontalImage;
+			}
+			if (imageDetails.getFlipVertical().booleanValue()) {
+				BufferedImage flippedVerticalImage = flipVertical(flippedImage);
+				flippedImage = null;
+				flippedImage = flippedVerticalImage;
+			}
+
+			BufferedImage rotatedImage = rotateImage(flippedImage, imageDetails.getRotation(),
+					imageDetails.getBackgroundColor(), actualFile);
+
+			BufferedImage croppedImage = rotatedImage;
+			if (imageDetails.getCropAreaWidth() > 0 && imageDetails.getCropAreaHeight() > 0) {
+				croppedImage = cropImage(rotatedImage, imageDetails.getCropAreaX(), imageDetails.getCropAreaY(),
+						imageDetails.getCropAreaWidth(), imageDetails.getCropAreaHeight());
+			}
+
+			return Mono.just(croppedImage);
+		} catch (IOException e) {
+			e.printStackTrace();
+			return Mono.empty();
+		}
+	}
+
+	private boolean checkIfNameExistsInFileSystem(String newImageName, File directory) {
+	    Path filePath = Paths.get(directory.getAbsolutePath(), newImageName);
+	    return Files.exists(filePath);
+	}
+
+    public BufferedImage rotateImage(BufferedImage originalImage, double angle, String backgroundColor, File file) {
+    	int width = originalImage.getWidth();
+        int height = originalImage.getHeight();
+        
+        double radians = Math.toRadians(angle);
+        double rotatedWidth = Math.abs(Math.sin(radians) * height) + Math.abs(Math.cos(radians) * width);
+        double rotatedHeight = Math.abs(Math.sin(radians) * width) + Math.abs(Math.cos(radians) * height);
+        
+        String fileExtension = getFileExtension(file);
+        BufferedImage rotatedImage;
+        
+        if(fileExtension.equals("png")) {
+        	rotatedImage = new BufferedImage((int) rotatedWidth, (int) rotatedHeight, BufferedImage.TYPE_INT_ARGB);
+        } else {
+        	rotatedImage = new BufferedImage((int) rotatedWidth, (int) rotatedHeight, originalImage.getType());
+        }
+
+        Graphics2D g2d = rotatedImage.createGraphics();
+        
+        Color color = getColorFromString(backgroundColor);
+        g2d.setColor(color);
+        g2d.fillRect(0, 0, rotatedImage.getWidth(), rotatedImage.getHeight());
+
+        AffineTransform transform = new AffineTransform();
+        transform.rotate(radians, rotatedWidth / 2, rotatedHeight / 2);
+
+        g2d.setTransform(transform);
+        int x = (int) ((rotatedWidth - width) / 2);
+        int y = (int) ((rotatedHeight - height) / 2);
+        g2d.drawImage(originalImage, x, y, null);
+        g2d.dispose();
+
+        return rotatedImage;
+    }
+	
+	public BufferedImage resizeImage(BufferedImage originalImage, int targetWidth, int targetHeight) throws IOException {
```

Comment:
here ioexception clause was not required as this method doesn't throw any error.